lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -510,6 +510,24 @@ static bool isOkToBitSelectFrom(Value v) {
   return false;
 }
 
+/// Find a nested IfOp in an else block that can be printed as `else if`
+/// instead of nesting it into a new `begin` - `end` block.  The block must
+/// contain a single IfOp and optionally expressions which can be hoisted out.
+static IfOp findNestedElseIf(Block *elseBlock) {
```

Comment:
Is isVerilogExpression enough?  Right now, some expressions might print as statements with synthesized wires (EmitExprAsStmt).

Reply:
That's fine - any additional wires are hoisted to the scope which contains the outermost if.